contracts/StablecoinConverter.sol

```diff
@@ -270,26 +276,23 @@ contract StablecoinConverter is EpochTokenLocker {
             updateRemainingOrder(owners[i], orderIds[i], executedSellAmount);
             addBalanceAndBlockWithdrawForThisBatch(owners[i], tokenIdToAddressMap(order.buyToken), executedBuyAmount);
         }
-        // doing all subtractions after all additions (in order to avoid negative values)
+        // Perform all subtractions after additions to avoid negative values
         for (uint256 i = 0; i < owners.length; i++) {
-            (, uint128 executedSellAmount) = getTradedAmounts(volumes[i], orders[owners[i]][orderIds[i]]);
-            subtractBalance(
-                owners[i],
-                tokenIdToAddressMap(orders[owners[i]][orderIds[i]].sellToken),
-                executedSellAmount
-            );
+            Order memory order = orders[owners[i]][orderIds[i]];
+            (, uint128 executedSellAmount) = getTradedAmounts(volumes[i], order);
+            subtractBalance(owners[i], tokenIdToAddressMap(order.sellToken), executedSellAmount);
         }
         uint256 disregardedUtility = 0;
         for (uint256 i = 0; i < owners.length; i++) {
             disregardedUtility = disregardedUtility.add(evaluateDisregardedUtility(orders[owners[i]][orderIds[i]], owners[i]));
         }
+        tokenConservation.checkTokenConservation();
```

Comment:
Is there a reason this had to be moved?

Reply:
@josojo requested it and I can't remember why anymore. Will get back to you if @josojo doesn't beat me to it.

Reply:
I asked for it, as otherwise, we would do the conversion int->uint, 
uint256(tokenConservation[0]) before the check that tokenConservation[0]>0

Reply:
😨 That's a very brittle and implicit dependency on code/function ordering. Would it make sense to safeCast instead?

Reply:
Unfortunately, the location of this is going to cause problems with the check that solutions are non-trivial.

Observe the following three lines of code:


```
        tokenConservation.checkTokenConservation();
        uint256 burntFees = uint256(tokenConservation[0]) / 2;
        require(utility.add(burntFees) > disregardedUtility, "Solution must be better than trivial");
```

We have a unit test which ensures trivial solution is rejected by the smart contract. However, the trivial solution has no fees and will, thus, fail the token conservation before the check for triviality.

My options here are the following:

- Remove test which ensures contract rejects trivial solution.
- Move the tokenConservation check till after the triviality check.